packages/pigeon/lib/gobject_generator.dart

```diff
@@ -0,0 +1,2071 @@
+    indent.writeln(' */');
+    indent.writeScoped('typedef enum {', '} $enumName;', () {
+      for (int i = 0; i < anEnum.members.length; i++) {
+        final EnumMember member = anEnum.members[i];
+        final String itemName =
+            _getEnumValue(dartPackageName, anEnum.name, member.name);
+        indent.writeln(
+            '$itemName = $i${i == anEnum.members.length - 1 ? '' : ','}');
+      }
+    });
+  }
+
+  @override
+  void writeDataClass(
+    GObjectOptions generatorOptions,
+    Root root,
+    Indent indent,
+    Class classDefinition, {
+    required String dartPackageName,
+  }) {
+    final String module = _getModule(generatorOptions, dartPackageName);
+    final String className = _getClassName(module, classDefinition.name);
+    final String methodPrefix = _getMethodPrefix(module, classDefinition.name);
+
+    indent.newln();
+    indent.writeln('/**');
+    indent.writeln(' * $className:');
+    indent.writeln(' *');
+    addDocumentationComments(
+        indent, classDefinition.documentationComments, _docCommentSpec);
+    indent.writeln(' */');
+
+    indent.newln();
+    _writeDeclareFinalType(indent, module, classDefinition.name);
+
+    indent.newln();
+    final List<String> constructorArgs = <String>[];
+    for (final NamedType field in classDefinition.fields) {
+      final String fieldName = _getFieldName(field.name);
+      final String type = _getType(module, field.type);
+      constructorArgs.add('$type $fieldName');
+      if (_isNumericListType(field.type)) {
+        constructorArgs.add('size_t ${fieldName}_length');
+      }
+    }
+    indent.writeln('/**');
+    indent.writeln(' * ${methodPrefix}_new:');
+    for (final NamedType field in classDefinition.fields) {
+      final String fieldName = _getFieldName(field.name);
+      indent.writeln(' * $fieldName: field in this object.');
+      if (_isNumericListType(field.type)) {
+        indent.writeln(' * ${fieldName}_length: length of @$fieldName.');
+      }
+    }
+    indent.writeln(' *');
+    indent.writeln(' * Creates a new #${classDefinition.name} object.');
+    indent.writeln(' *');
+    indent.writeln(' * Returns: a new #$className');
+    indent.writeln(' */');
+    indent.writeln(
+        "$className* ${methodPrefix}_new(${constructorArgs.join(', ')});");
+
+    for (final NamedType field in classDefinition.fields) {
+      final String fieldName = _getFieldName(field.name);
+      final String returnType = _getType(module, field.type);
+
+      indent.newln();
+      indent.writeln('/**');
+      indent.writeln(' * ${methodPrefix}_get_$fieldName');
+      indent.writeln(' * @object: a #$className.');
+      if (_isNumericListType(field.type)) {
+        indent
+            .writeln(' * @length: location to write the length of this value.');
+      }
+      indent.writeln(' *');
+      if (field.documentationComments.isNotEmpty) {
+        addDocumentationComments(
```

Comment:
Can these documentation comments be added using the `addDocumentationComments` method? If there's a reason not to, it's ok as is.

Reply:
This comment applies to any manually added documentation comments.

Reply:
I can also do it later myself if you don't want to deal with it :)

Reply:
I pushed a change to adopt `addDocumentationComment` everywhere; do you want to give it a quick review? (I validated that the output is identical to before, so just the code structure).